docs/src/snforge-advanced-features/backtrace.md

```diff
@@ -0,0 +1,70 @@
+# Backtrace
+
+## Prerequisites
+
+Backtrace feature relies on debug information provided by Scarb. To generate the necessary debug information, you need
+to have:
+
+1. [Scarb](https://github.com/software-mansion/scarb) version `2.8.0` or higher
+2. `Scarb.toml` file with the following Cairo compiler configuration:
+
+```toml
+[profile.dev.cairo]
+unstable-add-statements-code-locations-debug-info = true
+unstable-add-statements-functions-debug-info = true
+```
+
+## Usage
+
+> 📝 **Note**  
+> Currently, only the last line of failure in each contract is guaranteed to appear in the backtrace. The complete call
+> tree is not fully supported yet; however, in most cases, it will be available. It internally relies on the inlining
+> behavior of the compiler, and a full backtrace is available if all functions are inlined. To obtain a more detailed
+> backtrace, ensure that
+> your [inlining strategy](https://docs.swmansion.com/scarb/docs/reference/manifest.html#inlining-strategy) in
+`Scarb.toml` is set to `default`.
+
+When a contract call fails, the error message alone may not always provide enough information to identify the root cause
+of the issue. To aid in debugging, `snforge` offers a feature that can generate a backtrace of the execution.
+
+If your contract fails and a backtrace can be generated, `snforge` will prompt you to run the operation again with the
+`SNFORGE_BACKTRACE=1` environment variable (if it’s not already configured). For example, you may see failure data like
+this:
+
+```
+Failure data:
+    (0x454e545259504f494e545f4e4f545f464f554e44 ('ENTRYPOINT_NOT_FOUND'), 0x454e545259504f494e545f4641494c4544 ('ENTRYPOINT_FAILED'))
+note: run with `SNFORGE_BACKTRACE=1` environment variable to display a backtrace
+```
+
+To enable backtraces, simply set the `SNFORGE_BACKTRACE=1` environment variable and rerun the operation, as shown below:
+
+```bash
```

Comment:
```suggestion
```shell
```

Reply:
Also, please use the `<details>...</details>` syntax like we do for different snippets. 

See here for example https://github.com/foundry-rs/starknet-foundry/blob/ed614495ec404f9003e4325009cdd9ca08d13157/docs/src/starknet/call.md?plain=1#L28 or most other snippets in docs

Reply:
cc @franciszekjob

Reply:
Yup, let's rework this to have `snforge test` command and its output below, so the snippet will be tested (as in the ref above).